mne_connectivity/tests/test_wsmi.py

```diff
@@ -0,0 +1,582 @@
+    )
+
+    # All should have same shape and finite values
+    for conn in [conn_k3_t1, conn_k4_t1, conn_k3_t2]:
+        assert conn.get_data().shape == (n_epochs, 3)  # 3 choose 2 = 3 connections
+        assert np.all(np.isfinite(conn.get_data()))
+
+
+def test_wsmi_weighted_vs_unweighted():
+    """Test weighted parameter produces different results for wSMI vs SMI."""
+    sfreq = 100.0
+    n_epochs, n_times = 2, 200
+
+    # Create simple structured data
+    data = np.zeros((n_epochs, 3, n_times))
+    t = np.linspace(0, n_times / sfreq, n_times)
+
+    for epoch in range(n_epochs):
+        data[epoch, 0, :] = np.sin(2 * np.pi * 10 * t)
+        data[epoch, 1, :] = np.sin(2 * np.pi * 10 * t)  # Identical
+        data[epoch, 2, :] = np.sin(2 * np.pi * 12 * t)  # Different
+
+    ch_names = ["A", "B", "C"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Test wSMI vs SMI
+    conn_wsmi = wsmi(epochs, kernel=3, tau=1, weighted=True)
+    conn_smi = wsmi(epochs, kernel=3, tau=1, weighted=False)
+
+    # Should have different method names
+    assert conn_wsmi.method == "wSMI"
+    assert conn_smi.method == "SMI"
+
+    # Should produce different results due to weighting
+    assert not np.array_equal(conn_wsmi.get_data(), conn_smi.get_data()), (
+        "wSMI and SMI should produce different results"
+    )
+
+    # Both should have finite values
+    assert np.all(np.isfinite(conn_wsmi.get_data()))
+    assert np.all(np.isfinite(conn_smi.get_data()))
+
+
+def test_wsmi_basic_functionality():
+    """Test basic wSMI functionality and object properties."""
+    sfreq = 100.0
+    n_epochs, n_channels, n_times = 2, 4, 200
+
+    # Create deterministic data
+    t = np.linspace(0, n_times / sfreq, n_times)
+    data = np.zeros((n_epochs, n_channels, n_times))
+
+    for epoch in range(n_epochs):
+        data[epoch, 0, :] = np.sin(2 * np.pi * 10 * t)  # 10 Hz
+        data[epoch, 1, :] = np.sin(2 * np.pi * 12 * t)  # 12 Hz
+        data[epoch, 2, :] = np.sin(2 * np.pi * 15 * t)  # 15 Hz
+        data[epoch, 3, :] = np.sin(2 * np.pi * 8 * t)  # 8 Hz
+
+    ch_names = ["Fz", "Cz", "Pz", "Oz"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    montage = mne.channels.make_standard_montage("standard_1020")
+    epochs.set_montage(montage, on_missing="ignore")
+
+    # Test basic computation
+    conn = wsmi(epochs, kernel=3, tau=1)
+
+    # Check basic properties
+    assert conn.method == "wSMI"
+    assert conn.n_nodes == 4
+    assert conn.n_epochs_used == 2
+
+    # Check data shape and validity
+    data_matrix = conn.get_data()
+    expected_connections = 4 * 3 // 2  # 4 choose 2 = 6 connections
+    assert data_matrix.shape == (2, expected_connections)
+    assert np.all(np.isfinite(data_matrix))
+
+    # Test with different channel types (suppress unit change warning)
+    import warnings
```

Comment:
import should be at top of file...  but don't do it that way either.  Use `with pytest.warns(...)`